url4df616fce9_s3e0n1g1_prototype3_tgz-pJ8-Trading_Engine_MainJ8/src/Selecting_Algothrim/lecMSMomentum.java

```diff
@@ -0,0 +1,384 @@
+			}else if ( mode == SELL_MODE){
+				//tempCR = new CoupleReciept();
+				tempCR.setSellList(newReceiptList);
+				this.listOfAllReciept.add(tempCR);
+				mode = BUY_MODE;
+			}
+			else{
+				System.out.println("There is a problem");
+			}
+		}
+
+		//System.out.println("tempCR.getBuyList() count: " + tempCR.getBuyList().size());
+	}
+	
+	// should be changed
+	
+	
+	public LinkedList<CoupleReciept> getAllReciept(){
+		return this.listOfAllReciept;
+	}
+	
+	
+	
+	private double averageChange;
+	private double lastAverage;
+	private int shareQuantityToTrade;
+	private int shareQuantityLeft;
+	private int shareQuantity;
+	private LinkedList<Integer> sellReceipt = new LinkedList<Integer>(); 
+	private LinkedList<Integer> buyReceipt = new LinkedList<Integer>(); 
+	private double average;
+	private int mode;
+	private double[] tenArray = new double[10];
+	private  boolean recording;
+	private int counting;
+	private static final int SIZE_OF_ARRAY = 10;
+	private boolean arrayIsFull;
+	private static final int SELL_MODE = 1;
+	private static final int BUY_MODE = -1;
+	private  double epsilon = 0.001;
+	private double epsilonPrice = 0.0002;
+	private LinkedList<CoupleReciept> listOfAllReciept = new LinkedList<CoupleReciept>();
+
+	// NEW ADDED PART    week 12
+	long tempTime;// for calculation of time
+	long tempTime2;
+	Time tempAverageTime; 
+	double outComePercentage;
+	double tempDebit;
+	double tempCredit;
+	int tempNumberOfTrades;// this is for the reason of getting average time of trades
+	LinkedList<resultObjectL> listOfResultsFromTrade = new LinkedList<resultObjectL>();
+	//////////////////
+	
+	
+	private CoupleReciept tempCR;
+
+	
+	
+	
+	/// NEW methode
+	public void getSTrade( LinkedList< ResultData> listOfTrades){
+		
+		//ResultData tempRD;
+		
+		if ( mode == BUY_MODE ){// means these trades are because of our Buy
+			tempTime  = 0;
+			tempDebit = 0;
+			tempNumberOfTrades = 0;
+			for ( ResultData tempRD : listOfTrades){
+				tempDebit += tempRD.getPrice();
+				tempTime += tempRD.getTime().getTime();
+			}
+			tempNumberOfTrades = listOfTrades.size();
+		}else if ( mode == SELL_MODE){
+			tempCredit = 0;
+			for ( ResultData tempRD : listOfTrades){
+				tempCredit += tempRD.getPrice();
+				tempTime += tempRD.getTime().getTime();
+			}
+			// when we sell couple gets complete
+			tempNumberOfTrades += listOfTrades.size();
+			tempTime2 = (long) tempTime/tempNumberOfTrades;
+			tempAverageTime = new Time(tempTime2);
+			outComePercentage = (tempCredit - tempDebit)/(tempCredit );
+			listOfResultsFromTrade.add(new resultObjectL(  tempAverageTime, outComePercentage));
+		}
+		else{
+			System.out.println("There is a problem on getSTrade Methode on LecMSMomentum");
+		}
+		
+		
+		
+		
+		
+		
+	}
+	/**
+	 * it's also for double checking
+	 */
+	public LinkedList< resultObjectL> getResultListFromStrategy(){
+		return listOfResultsFromTrade;
+	}
+	
+	
+	
+	public LinkedList<signalObject> generateSignalList(MyBidList myBidList,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.